Sources/PushNotifications.swift

```diff
@@ -211,24 +211,26 @@ import Foundation
      - Precondition: `deviceToken` should not be nil.
      */
     /// - Tag: registerDeviceToken
-    @objc public func registerDeviceToken(_ deviceToken: Data, completion: @escaping () -> Void = {}) {
+    @objc public func registerDeviceToken(_ deviceToken: Data, completion: @escaping (Error?) -> Void = { _ in }) {
         guard
             let instanceId = Instance.getInstanceId(),
             let url = URL(string: "https://\(instanceId).pushnotifications.pusher.com/device_api/v1/instances/\(instanceId)/devices/apns")
         else {
-            print("[Push Notifications] - Something went wrong. Please check your instance id: \(String(describing: Instance.getInstanceId()))")
+            completion(PushNotificationsError.error("[Push Notifications] - Something went wrong. Please check your instance id: \(String(describing: Instance.getInstanceId()))"))
             return
         }
 
         self.deviceToken = deviceToken
 
         if Device.idAlreadyPresent() {
             // TODO: Handle the token change.
+            completion(PushNotificationsError.error("Device is already present"))
             return
         }
 
         networkService.register(url: url, deviceToken: deviceToken, instanceId: instanceId) { [weak self] result in
             guard let strongSelf = self else {
+                completion(PushNotificationsError.error("deallocated"))
```

Comment:
Maybe we want something more explicit here. I'm not sure when this can happen either :/

Reply:
Agreed, this isn’t an error either, but it was another place where the completion should be called, IMO.